### ISSUE TITLE
Terraform destroy does not handle 404 on Google Compute Instance

### ISSUE THREAD

**Issue by clstokes** [hashicorp/terraform#3990]
If I create a Google Compute Instance with Terraform, then remove the instance via the Google Cloud console, a `terraform destroy` fails with a _no longer exists_ error (likely a 404) and won't continue. Other Terraform providers seem to handle this case by handling the 404 as a successful destroy and continuing on.

Tested with Terraform **v0.6.6**. 

Terraform config:
```
resource "google_compute_instance" "main" {
  name         = "main"
  machine_type = "n1-standard-1"
  zone         = "us-central1-f"

  disk {
    image = "ubuntu-1404-trusty-v20151113"
  }

  network_interface {
    network = "default"
    access_config {
      // Ephemeral IP
    }
  }
}
```

Abridged Terraform output...
```
$ terraform apply

...

google_compute_instance.main: Creating...
  can_ip_forward:                             "" => "0"
  disk.#:                                     "" => "1"
  disk.0.auto_delete:                         "" => "1"
  disk.0.image:                               "" => "ubuntu-1404-trusty-v20151113"
  machine_type:                               "" => "n1-standard-1"
  metadata_fingerprint:                       "" => "<computed>"
  name:                                       "" => "main"
  network_interface.#:                        "" => "1"
  network_interface.0.access_config.#:        "" => "1"
  network_interface.0.access_config.0.nat_ip: "" => "<computed>"
  network_interface.0.address:                "" => "<computed>"
  network_interface.0.name:                   "" => "<computed>"
  network_interface.0.network:                "" => "default"
  self_link:                                  "" => "<computed>"
  tags_fingerprint:                           "" => "<computed>"
  zone:                                       "" => "us-central1-f"
google_compute_instance.main: Creation complete

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

...

$ 
$ # Instance removed via Google Cloud console
$ 
$ terraform destroy -force

...

google_compute_instance.main: Refreshing state... (ID: main)
Error refreshing state: 1 error(s) occurred:

* google_compute_instance.main: Resource hashicorp-clstokes no longer exists
$
```

**Comment by lwander** [comment]
This has already been brought up in #3742, and is being addressed in #3913. However, @apparentlymart made the case that the typical behavior is to allow `terraform show` to resolve this by calling `Read(..)` on the deleted resources, marking them as no longer existing. What do you think, @clstokes?

**Comment by clstokes** [comment]
I'm definitely good with @apparentlymart's suggestion. I'll close this in favor of #3742. Thanks @lwander!
